### PR TITLE
Compat for Julia #18510, #18484

### PR DESCRIPTION
**Pull request by davidagold**
Compatibility for https://github.com/JuliaLang/julia/pull/18510 and https://github.com/JuliaLang/julia/pull/18484.

@tkelman c.f. https://github.com/davidagold/StructuredQueries.jl/commit/a8ede1791ef3fa1d2e2c1478728ba807459bd973#commitcomment-19450532

### CODE REVIEW

**Comment by tkelman** [comment]
thanks, add to README

**Comment by davidagold** [comment]
Thanks! cc @nalimilan who may be interested

**Comment by davidagold** [comment]
@tkelman how many reviews does this need?

File: src/Compat.jl [hunk 1]
```diff
@@ -613,6 +613,8 @@ function _compat(ex::Expr)
         elseif VERSION < v"0.5.0-dev+4340" && length(ex.args) > 3 &&
                 istopsymbol(withincurly(ex.args[1]), :Base, :show)
             ex = rewrite_show(ex)
+        elseif VERSION < v"0.6.0-dev.826" && f === :Nullable && length(ex.args) == 3 # julia#18510
+            ex = Expr(:call, :Nullable, ex.args[2], Expr(:call, :(Compat._Nullable_field2), ex.args[3]))
```
Comment (TotalVerb) [inline src/Compat.jl]: This does not handle `Nullable{Int}(x, b)` properly. See how `show` is handled above (need `withincurly` and `istopsymbol`.

**Comment by davidagold** [comment]
Good catch -- thanks, @TotalVerb . I've made those changes and added tests.

File: src/Compat.jl [hunk 2]
```diff
@@ -613,6 +613,9 @@ function _compat(ex::Expr)
         elseif VERSION < v"0.5.0-dev+4340" && length(ex.args) > 3 &&
                 istopsymbol(withincurly(ex.args[1]), :Base, :show)
             ex = rewrite_show(ex)
+        elseif VERSION < v"0.6.0-dev.826" && length(ex.args) == 3 && # julia#18510
+                istopsymbol(withincurly(ex.args[1]), :Base, :Nullable)
+            ex = Expr(:call, :Nullable, ex.args[2], Expr(:call, :(Compat._Nullable_field2), ex.args[3]))
```
Comment (TotalVerb) [inline src/Compat.jl]: `Expr(:call, f` or `Expr(:call, ex.args[1]`. Currently it doesn't behave properly for `Nullable{Any}(1.0, false)`.

**Comment by davidagold** [comment]
Thanks again for your thoroughness

**Comment by TotalVerb** [comment]
@davidagold I added two test cases for `eltype` of `Nullable{Any}`. I think this is good to merge now.

**Comment by tkelman** [comment]
Assuming no-one objects to tagging this. https://github.com/JuliaLang/METADATA.jl/pull/6791

**Pull request merged**